plutus-benchmark/script-contexts/test/V1/9.6/checkScriptContext1.pir.golden

```diff
@@ -47,8 +47,10 @@
            (d : data) ->
             let
               !tup : pair integer (list data) = unConstrData d
-              !index : integer = fstPair {integer} {list data} tup
-              !args : list data = sndPair {integer} {list data} tup
+              !index : integer
+                = case integer tup [(\(l : integer) (r : list data) -> l)]
+              !args : list data
+                = case (list data) tup [(\(l : integer) (r : list data) -> r)]
```

Comment:
That's one place where instead of binding those two separately, we should just pull the `case` to the outside and bind `index` and `args` in its branch. There's many more of these below.

If you grep for `BI.fst`, you'll find plenty of places where it appears together with `BI.snd`. Turning those into a single case would be a good optimization.